### PR TITLE
Drop `execa` dependency

### PR DESCRIPTION
**Pull request by fregante**
- [x] Test it in a real project because the tests actually skip the new shim altogether

I know this is probably not something welcome, but I looked at the usage of `execa` and maybe it's not necessary here. It'd make `os-locale` quite small

Screen Shot 1

### CODE REVIEW

**Comment by sindresorhus** [comment]
I'm ok with it. Just try to ensure there are no regressions.

File: execa.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Mini wrapper around child_process to make it behave a little like execa
+
+const {promisify} = require('util');
+const childProcess = require('child_process');
+
+module.exports = promisify(childProcess.execFile).then(value => {
```
Comment (sindresorhus) [inline execa.js:6]: Use async keyword over `.then`.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by fregante** [comment]
> I'm ok with it. Just try to ensure there are no regressions.

Good idea, the tests here don’t really test the execa shim.

If you point me to a project(s) that uses os-locale extensively with cross-platform tests I can try plugging it in an sending a test PR on those repos to trigger the tests.

File: execa.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const childProcess = require('child_process');
+
+const execFile = promisify(childProcess.execFile);
+
+/**
+@param  {string} command
+@param  {string[]} args
+
+@returns Promise<import('child_process').ChildProcess>
+*/
```
Comment (fregante) [inline execa.js:13]: I added these to help their use in editors that show types

Screen Shot

File: test.js [hunk 3]
```diff
@@ -5,6 +5,8 @@ const proxyquire = proxyquireBase.noPreserveCache().noCallThru();
 
 const expectedFallback = 'en-US';
 
+const execaImport = './execa';
```
Comment (brodybits) [inline test.js:8]: Couldn't you use `execa` from devDependencies just for testing?

Comment (fregante) [reply test.js:8]: This file isn't actually importing any modules. `proxyquire` mocks the import that has this path in `index.js`

**Comment by sindresorhus** [comment]
Bump :)

**Comment by fregante** [comment]
> If you point me to a project(s) that uses os-locale extensively with cross-platform tests I can try plugging it in an sending a test PR on those repos to trigger the tests.

Hey I was gonna bump 👆 ! 😃

**Comment by sindresorhus** [comment]
> If you point me to a project(s) that uses os-locale extensively with cross-platform tests I can try plugging it in an sending a test PR on those repos to trigger the tests.

I don't actually know. https://www.npmjs.com/browse/depended/os-locale is completely useless as you cannot sort by popularity...

**Comment by fregante** [comment]
> I don't actually know. [npmjs.com/browse/depended/os-locale](https://www.npmjs.com/browse/depended/os-locale) is completely useless as you cannot sort by popularity...

```sh
npx dependents --package os-locale  --sort downloads --total 25 --json
```

```
[
  {
    'strong-globalize': {
      downloads: 195109,
      forks: 14,
      stars: 22,
      watchers: 37,
      url: 'github.com/strongloop/strong-globalize'
    }
  },
  {
    'addons-linter': {
      downloads: 33243,
      forks: 111,
      stars: 216,
      watchers: 38,
      url: 'github.com/mozilla/addons-linter'
    }
  },
  {
    'generator-jhipster': {
      downloads: 30877,
      forks: 3580,
      stars: 18653,
      watchers: 788,
      url: 'github.com/jhipster/generator-jhipster'
    }
  },
  {
    '@umijs/fabric': {
      downloads: 19671,
      forks: 60,
      stars: 324,
      watchers: 7,
      url: 'github.com/umijs/fabric'
    }
  },
  {
    'gulp-reporter': {
      downloads: 19230,
      forks: 2,
      stars: 17,
      watchers: 1,
      url: 'github.com/gucong3000/gulp-reporter'
    }
  },
  {
    eclint: {
      downloads: 19191,
      forks: 20,
      stars: 291,
      watchers: 12,
      url: 'github.com/jedmao/eclint'
    }
  },
  {
    'umi-core': {
      downloads: 6103,
      forks: 1719,
      stars: 10880,
      watchers: 263,
      url: 'github.com/umijs/umi'
    }
  },
  {
    hint: {
      downloads: 4607,
      forks: 286,
      stars: 3182,
      watchers: 70,
      url: 'github.com/webhintio/hint'
    }
  },
  {
    'cheerio-httpcli': {
      downloads: 3280,
      forks: 24,
      stars: 247,
      watchers: 10,
      url: 'github.com/ktty1220/cheerio-httpcli'
    }
  },
  {
    '@boost/translate': {
      downloads: 2298,
      forks: 2,
      stars: 49,
      watchers: 1,
      url: 'github.com/milesj/boost'
    }
  }
```

**Comment by fregante** [comment]
Tested:

- [x] strong-globalize

Methodology:

1. `npm test` succeeds
1. Replace `'execa'` with `'./execa'` in `os-locale` dependency
2. `npm test` fails
3. Add `./execa.js`
1. `npm test` succeeds

**Comment by fregante** [comment]
Thanks!

**Pull request merged**